### DISCUSSION TITLE
Transform `BigInt` fields to `number` or `string` globally (maybe via Prisma Client extensions)?

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#20188] [Q&A] [answered]
Hey!
I have `BigInt` fields in my schema (PostgreSQL) which are not nice to use in React code.
The workarounds I've seen all seem hacky to me and are only ad-hoc solutions when using the data (instead of on a global level).

Now that Prisma Client extensions are out, could one write an extension that automatically transform all fields in the schema defined with `BigInt` to either `number` or `string`? If so, how would it look like?

**Comment by ludralph** [comment]
Hi @bennettdams 👋 

You could write an extension to transform the fields from BigInt to either number or string. You can take a look at this [example](https://www.prisma.io/blog/client-extensions-preview-8t3w27xkrxxn#example-transformed-fields) for transforming fields using Prisma Client extension.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by bennettdams** [reply]
Thanks for the answer and the examples.
In my case, the fields which are of type `BigInt` all have different names and live in different models.
I thought it would be great to do something like: 

"if the field is of type `BigInt` (in the Prisma schema), transform it to `number`/`string`".

Skimming through the examples, I have not seen a way to access the "Prisma type (`BigInt`)" which I could use to filter. Is there another way to transform all `BigInt`s globally?

**Reply by Crinchy** [reply, marked as answer]
Hey bro, i am currently using this workaround, to fix the big int problem globally, i have it in my index.ts.
```
(BigInt.prototype as any).toJSON = function () {
  return Number(this)
};
```
